libglusterfs/src/inode.c

```diff
@@ -1602,6 +1603,7 @@ inode_table_prune(inode_table_t *table)
             }
 
             table->lru_size--;
+            entry->in_lru_list = _gf_false;
```

Comment:
After line 1586, I would check that `entry->in_lru_list` is true, but now I don't understand the code in lines 1592..1596.  The `GF_ASSERT` in line 1593 should be removed because it will always be false, and the `list_move_tail()` only moves the entry from the head to the tail of the same list, so the increment in `table->lru_size` in line 1596 seems incorrect. In this case I'm also not sure if decrementing `lru_size` at line 1584 is correct.

Can you check all this ?

Reply:
Yes you are correct the increment at line 1596 is not require but at line 1584 lru_size decrement is required to come out from a loop.

Reply:
Oh, `lru_size` is used to ensure that we only traverse each element once here... in that case it's fine.